### PR TITLE
INT-5705 upgrade beta apis for iq

### PR DESCRIPTION
**Pull request by jflinchbaugh**
update Ingress to v1 for IQ.

JIRA: https://issues.sonatype.org/browse/INT-5705
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/INT-5705-upgrade-beta-apis-for-iq/

### CODE REVIEW

File: charts/nexus-iq/README.md [hunk 1]
```diff
@@ -4,19 +4,21 @@
 
 ### Prerequisites
 
-- Kubernetes 1.8+ with Beta APIs enabled
+- Kubernetes 1.19+
```
Comment (jflinchbaugh) [inline charts/nexus-iq/README.md:7]: They're up to k8s 1.23.0, 1.22 removed support for beta ingress in favor of full v1.

File: charts/nexus-iq/templates/ingress.yaml [hunk 2]
```diff
@@ -2,11 +2,7 @@
 {{- $fullName := include "iqserver.fullname" . -}}
 {{- $svcPort := .Values.service.port -}}
 {{- $ingressPath := .Values.ingress.path -}}
-{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
-apiVersion: extensions/v1beta1
-{{- end }}
+apiVersion: networking.k8s.io/v1
```
Comment (jflinchbaugh) [inline charts/nexus-iq/templates/ingress.yaml:5]: 1.14 is pretty old, so i'm not going to try to support those old versions.

File: charts/nexus-iq/templates/ingress.yaml [hunk 3]
```diff
@@ -37,14 +33,20 @@ spec:
...
         paths:
           - path: {{ .Values.ingress.hostUIPath }}
+            pathType: Prefix
             backend:
-              serviceName: {{ $fullName }}
-              servicePort: 8070
+              service:
+                name: {{ $fullName }}
+                port:
+                  number: 8070
```
Comment (jflinchbaugh) [inline charts/nexus-iq/templates/ingress.yaml:41]: this is the structure of the new ingress/v1 object.

File: charts/nexus-iq/values.yaml [hunk 4]
```diff
@@ -90,19 +90,17 @@ service:
...
-  tls: []
-    # - secretName: nexus-local-tls
-    #   hosts:
-    #     - iqserver.local
-    #     - admin.iqserver.local
+  # tls:
+  #   - secretName: nexus-local-tls
+  #     hosts:
+  #       - iq-server.demo
+  #       - admin.iq-server.demo
```
Comment (jflinchbaugh) [inline charts/nexus-iq/values.yaml:103]: these hosts should have matched the host names used for the endpoints, so now they do.

File: charts/nexus-iq/tests/ingress_test.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      ingress:
+        enabled: true
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: Ingress
+      - equal:
+          path: apiVersion
+          value: networking.k8s.io/v1
```
Comment (adrianpowell) [inline charts/nexus-iq/tests/ingress_test.yaml:16]: nice to see some sort of a test! Wild.

**Review by adrianpowell — APPROVED** [review]

File: charts/nexus-iq/tests/ingress_test.yaml (hunk 5 above)
Comment (hectorhuol) [reply charts/nexus-iq/tests/ingress_test.yaml:16]: This will help us a lot! Thanks for putting all of this together.

**Review by hectorhuol — APPROVED** [review]
+1

**Pull request merged**